contracts/src/Teleporter/upgrades/README.md

```diff
@@ -137,18 +127,69 @@ Using specific version:
+
+#### Updating `minTeleporterVersion`
+
+The `TeleporterUpgradeable.updateMinTeleporterVersion` function updates the `minTeleporterVersion` used to check which Teleporter versions can deliver messages to the dApp, and emits the `MinTeleporterVersionUpdated` event. The `updateMinTeleporterVersion` function should **ONLY** be called by the dApp when it completes delivery of messages from the old Teleporter contract, and now wants to update the `minTeleporterVersion` to only allow the new Teleporter version. By default, `updateMinTeleporterVersion` can only be called with a version greater than the current `minTeleporterVersion` and less than `latestVersion` in the Teleporter registry. So once this function is called, the dApp will no longer be able to receive messages from the old Teleporter contract version, unless the old version's Teleporter address was registered in the registry again with a new version.
+
+> Example: Update the minimum Teleporter version to 2
+>```
+>cast send <DAPP_ADDRESS> "updateMinTeleporterVersion(uint256)" 2 
+>```
+
+### Pausing Teleporter version interactions
+
+Dapps that inherit from `TeleporterUpgradeable` can pause Teleporter interactions by calling `TeleporterUpgradeable.pauseTeleporterAddress`. This function prevents the dApp contract from interacting with the paused Teleporter address when sending or receiving Teleporter messages.
 
 `pauseTeleporterAddress` can only be called by addresses with the dApp's upgrade access, checked through `TeleporterUpgradeable._checkTeleporterUpgradeAccess`.
 
+The Teleporter address corresponding to a Teleporter version can be fetched from the registry with `TeleporterRegistry.getAddressFromVersion`
+
+> Example: Pause Teleporter version 3
+>```
+>versionThreeAddress=$(cast call <REGISTRY_ADDRESS> "getAddressFromVersion(uint256)(address)" 3)
+>cast send <DAPP_ADDRESS> "pauseTeleporterAddress(address)" $versionThreeAddress
+>```
+
+#### Pause all Teleporter interactions
+
+To pause all Teleporter interactions, `TeleporterUpgradeable.pauseTeleporterAddress` must be called for every Teleporter version from the `minTeleporterVersion` to the latest Teleporter version registered in `TeleporterRegistry`. The latest Teleporter version can be obtained by inspecting the public variable `TeleporterRegistry.latestVersion`. The `minTeleporterVersion` can be obtained by calling `TeleporterUpgradeable.getMinTeleporterVersion`.
+
+> Example: Pause all registered Teleporter versions
+>```
+># Fetch the minimum Teleporter version
+>minVersion=$(cast call <DAPP_ADDRESS> "getMinTeleporterVersion()(uint256)")
+>
+># Fetch the latest registered version
+>latestVersion=$(cast call <REGISTRY_ADDRESS> "latestVersion()(uint256)")
+>
+># Pause all registered versions
+>for ((version=minVersion; version<=latestVersion; version++))
+>do
+>  versionAddress=$(cast call <REGISTRY_ADDRESS> "getAddressFromVersion(uint256)(address)" $version)
```

Comment:
I think it's possible for there to be gaps in version numbers right? If so, we should continue here if the version address isn't registered.

Reply:
Modified the example script, and added a note in the above description.